vendor/k8s.io/kubernetes/staging/src/k8s.io/apiextensions-apiserver/pkg/controller/openapi/conversion.go

```diff
@@ -33,9 +33,10 @@ func ConvertJSONSchemaPropsToOpenAPIv2Schema(in *apiextensions.JSONSchemaProps)
 	}
 
 	// dirty hack to temporarily set the type at the root. See continuation at the func bottom.
-	// TODO: remove for Kubernetes 1.15
 	oldRootType := in.Type
 	if len(in.Type) == 0 {
+		shallowClone := *in
+		in = &shallowClone
```

Comment:
have no idea why we didn't do that in the first place. But I remember a discussion, but not the arguments.